Shared/alerts/AlertConditionsController.cpp

```diff
@@ -606,15 +610,29 @@ void AlertConditionsController::onLayersChanged()
         continue;
 
       FeatureLayer* featLayer = qobject_cast<FeatureLayer*>(lyr);
-      if (!featLayer)
+      DynamicEntityLayer *dynamicEntityLayer = qobject_cast<DynamicEntityLayer*>(lyr);
+      if (!featLayer && !dynamicEntityLayer)
         continue;
 
-      if (featLayer->loadStatus() != LoadStatus::Loaded)
-        connect(featLayer, &FeatureLayer::doneLoading, this, &AlertConditionsController::onLayersChanged);
-      else
+      if (featLayer)
       {
-        newTargetList.append(featLayer->name());
-        existingLayerIds.append(featLayer->name());
+        if (featLayer->loadStatus() != LoadStatus::Loaded)
+          connect(featLayer, &FeatureLayer::doneLoading, this, &AlertConditionsController::onLayersChanged);
+        else
+        {
+          newTargetList.append(featLayer->name());
+          existingLayerIds.append(featLayer->name());
+        }
+      }
+      if (dynamicEntityLayer)
```

Comment:
I recommend some minor cleanup here. Rather than qobject_cast'int to both types upfront, check them sequentially in if-else cases. It can't be both. I would also use if-assignment checks to reduce extra scope.

Here's an example
```c++
if (FeatureLayer* featLayer = qobject_cast<FeatureLayer*>(lyr); featLayer)
{
  // do feature layer stuff
}
else if (DynamicEntityLayer* dynamicEntityLayer = qobject_cast<DynamicEntityLayer*>(lyr); dynamicEntityLayer)
{
  // do DynamicEntityLayer stuff
}
else 
{
  continue; // is this still needed here? Please review
}
```

Reply:
I replaced this block as well as another very similar instance later in the code.